policies/observability/PodMonitor.yaml

```diff
@@ -6,47 +6,79 @@ spec:
   rules:
     - name: configure-pod-monitor-labelling-schema
       match:
-        resources:
-          kinds:
-          - monitoring.coreos.com/v1/PodMonitor
+        any:
+        - resources:
+            kinds:
+            - monitoring.coreos.com/v1/PodMonitor
       mutate:
-        patchStrategicMerge:
-          spec:
-            podMetricsEndpoints:
-              - relabelings:
-                  # Pod Monitors are currently only supported on management clusters, so we can enforce the cluster_id and cluster_type.
-                - replacement: "[[ .Values.managementCluster.name ]]"
-                  targetLabel: cluster_id
-                - replacement: "management_cluster"
-                  targetLabel: cluster_type
-                  # Since we only support management clusters, all resources are considered as highest service priority
-                - replacement: "highest"
-                  targetLabel: service_priority
-                - replacement: "[[ .Values.provider.kind ]]"
-                  targetLabel: provider
-                - replacement: "[[ .Values.managementCluster.name ]]"
-                  targetLabel: installation
-                - sourceLabels: [__meta_kubernetes_namespace]
-                  targetLabel: namespace
-                - sourceLabels: [__meta_kubernetes_pod_label_app_kubernetes_io_name]
-                  targetLabel: app
-                - sourceLabels: [__meta_kubernetes_pod_label_app_kubernetes_io_instance]
-                  targetLabel: instance
-                - sourceLabels: [__meta_kubernetes_pod_name]
-                  targetLabel: pod
-                - sourceLabels: [__meta_kubernetes_pod_container_name]
-                  targetLabel: container
-                - sourceLabels: [__meta_kubernetes_pod_node_name]
-                  targetLabel: node
-                - sourceLabels: [__meta_kubernetes_node_label_role]
-                  targetLabel: role
-                - replacement: "[[ .Values.managementCluster.customer ]]"
-                  targetLabel: customer
+        foreach:
+        - list: request.object.spec.podMetricsEndpoints
+          patchesJson6902: |-
+            # Pod Monitors are currently only supported on management clusters, so we can enforce the cluster_id and cluster_type.
+            - path: "/spec/podMetricsEndpoints/{{elementIndex}}/relabelings/-1"
+              value: |-
+                {
+                  "replacement": "[[ .Values.managementCluster.name ]]",
+                  "targetLabel": "cluster_id"
+                },
+                {
+                  "replacement": "management_cluster",
+                  "targetLabel": "cluster_type"
+                },
+                # Since we only support management clusters, all resources are considered as highest service priority
+                {
+                  "replacement": "highest",
+                  "targetLabel": "service_priority"
```

Comment:
It's not a big change but can you set the priority to highest only if the value is not set before ? Using the regex: nil or something similar ?

Reply:
It's a separate change IMO.
* could change some service and pod's `service_priority`
* testing is more complex, it requires a metrics source with specific labels

So it's a bit time consuming, I'd rather do it later / in a separate PR.